### PR TITLE
Feature/single ride

### PR DESCRIPTION
**Pull request by felixportuphy**

### CODE REVIEW

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+      }
+      cy.get('a[href*="/your-rides"]').click()
+    })
+
+    afterEach(() => {
+      cy.logOut()
+      cy.clearCookies()
+    })
+
+    it('Navigates through Tabs', () => {
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:76]: test case name not descriptive enough

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      cy.dataCy('distance').should('have.css', 'color', 'rgb(66, 181, 70)')
+      cy.dataCy('duration').click()
+      cy.dataCy('duration').should('have.css', 'color', 'rgb(66, 181, 70)')
+    })
+
+    it('Filters Rides by "Recent"', () => {
+      cy.dataCy('recent').click()
+
+      cy.wait('@fetchedRides').then((res) => {
+        let jsonData = res.response
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:96]: I don't think there's a need for a variable declaration here. just do everything in the assertion.

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        expect(jsonData.statusCode).to.eq(200)
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:97]:
```suggestion
        expect(res.response.statusCode).to.eq(200)
```

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        let total = jsonData.body.total
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:98]:
```suggestion
        let total = res.response.body.total
```

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+        let firstRide = jsonData.body.data[0]
+        let lastRide = jsonData.body.data[total - 1]
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:100]: make the necessary changes for firstRide and lastRide too

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+        cy.get('.ae-ride-card').should('have.length', total)
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:101]: I'm not convinced with this assertion. It is not measured against anything. i.e this assertion could always pass.

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+    it('Filters Rides by "Old"', () => {
+      cy.dataCy('recent').dblclick()
+      cy.wait('@fetchedRides').then((res) => {
+        let jsonData = res.response
+        expect(jsonData.statusCode).to.eq(200)
+        let total = jsonData.body.total
+        let firstRide = jsonData.body.data[total - 1]
+        let lastRide = jsonData.body.data[0]
+        cy.get('.ae-ride-card').should('have.length', total)
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:114]: see above comments

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+
+    it('Filters "Favorites" rides in ascending order', () => {
+      cy.dataCy('favorites').click()
+      cy.wait('@favoritedRides').then((res) => {
+        let jsonData = res.response
+        expect(jsonData.statusCode).to.eq(200)
+        let total = jsonData.body.total
+        let firstRide = jsonData.body.data[total - 1]
+        let lastRide = jsonData.body.data[0]
+        cy.get('.ae-ride-card').should('have.length', total)
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:127]: see above comments

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+ *
+ */
+import { defaultDevices } from 'cypress/integration/_helpers/default-devices'
+import { baseUrl } from 'cypress/fixtures/shared/environment.js'
+import { loginUsers } from '../../../fixtures/amali-login.data'
+
+const testSpecName = 'amali-yourrides'
+const targetUrl = `${baseUrl}`
+const { exist_email, password } = loginUsers
+let firstRid, lastRid
```
Comment (JerryHukpati) [inline cypress/integration/tests/Rides/amali-filter-rides.spec.js:11]: I'm surprised you didn't rename these variables

File: cypress/integration/_helpers/update-profile.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+  cy.get('[data-cy="country"]')
+    .click({ force: true })
+    .scrollTo('bottom', { ensureScrollable: false, duration: 5000 })
+  cy.get('.v-list-item__title').contains('Bahrain').click()
+
+  //Enter social media links / number
+
+  cy.get('[data-cy="facebook"]').clear().type(`${facebook}/${firstName}${lastName}`)
+
```
Comment (JerryHukpati) [inline cypress/integration/_helpers/update-profile.js:59]: too many unnecessary lines in this block. the comment on line 56 as well I don't see it use

File: cypress/integration/_helpers/filter-rides.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+export const filterRides = (factor, order) => {
+  return cy
+    .intercept('POST', '**/statistics/all-user-rides', (req) => {
+      req.body = {
+        factor: factor,
+        order: order,
+      }
+    })
+    .as('fetchedRides')
+}
```
Comment (JerryHukpati) [inline cypress/integration/_helpers/filter-rides.js:10]: I don't think you are using this at the moment. you can delete it for now.

File: cypress/integration/_helpers/date.js [hunk 12]
```diff
@@ -1 +1,8 @@
 export const dateFormat = 'YYYY-MMM-DD_hh_mm'
+
+export const getTimeStamp = (rideDate) => {
+  const dateStr = rideDate
+  const date = new Date(dateStr.toString())
+  const timeStampMs = date.getTime()
```
Comment (JerryHukpati) [inline cypress/integration/_helpers/date.js:6]: timeStampMs??? consider better naming of variables. e.g timeInMilliSeconds, dateString

File: cypress-open.json [hunk 13]
```diff
@@ -8,12 +8,11 @@
...
+    "mode-run": false,
+    "MAILOSAUR_API_KEY": "7bQeUQP2jbtFuPvl"
   },
   "retries": 2,
   "scrollBehavior": "center",
   "chromeWebSecurity": false,
-  "env": {
-    "MAILOSAUR_API_KEY": "7bQeUQP2jbtFuPvl"
-  }
+  "watchForFileChanges": false
```
Comment (JerryHukpati) [inline cypress-open.json:17]: don't forget to remove this please.

**Review by JerryHukpati — CHANGES_REQUESTED** [review]
clean up the filter rides file based on the comments I made. I also see some console logs still in the file and also the comments are not necessary. If you do all these changes, I'm sure you'll be able to reduce the number of lines.

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js (hunk 1 above)
Comment (felixportuphy) [reply cypress/integration/tests/Rides/amali-filter-rides.spec.js]: Resolved

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js (hunk 2 above)
Comment (felixportuphy) [reply cypress/integration/tests/Rides/amali-filter-rides.spec.js]: Done

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js (hunk 3 above)
Comment (felixportuphy) [reply cypress/integration/tests/Rides/amali-filter-rides.spec.js]: Done

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js (hunk 4 above)
Comment (felixportuphy) [reply cypress/integration/tests/Rides/amali-filter-rides.spec.js]: Done

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js (hunk 5 above)
Comment (felixportuphy) [reply cypress/integration/tests/Rides/amali-filter-rides.spec.js]: Done

File: cypress/integration/tests/Rides/amali-filter-rides.spec.js (hunk 9 above)
Comment (felixportuphy) [reply cypress/integration/tests/Rides/amali-filter-rides.spec.js]: Done

File: cypress/integration/_helpers/update-profile.js (hunk 10 above)
Comment (felixportuphy) [reply cypress/integration/_helpers/update-profile.js]: Resolved

File: cypress/integration/_helpers/filter-rides.js (hunk 11 above)
Comment (felixportuphy) [reply cypress/integration/_helpers/filter-rides.js]: Done

File: cypress/integration/_helpers/date.js (hunk 12 above)
Comment (felixportuphy) [reply cypress/integration/_helpers/date.js]: Done

File: cypress-open.json (hunk 13 above)
Comment (felixportuphy) [reply cypress-open.json]: Done

**Pull request closed without merging**